test/e2e/secrets_test.go

```diff
@@ -128,6 +129,27 @@ func testUserDataTamper(t *testing.T) {
 	}
 }
 
+// testPrivateKeyChangeBYOH changes the SSH private key and ensures BYOH nodes are properly patched using the new key
+func testPrivateKeyChangeBYOH(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	t.Run("Patch BYOH node annotations on private key change", func(t *testing.T) {
+		// This test cannot be run on vSphere because this random key is not part of the vSphere template image.
+		if tc.CloudProvider.GetType() == config.VSpherePlatformType {
+			t.Skipf("Skipping for %s", tc.CloudProvider.GetType())
+		}
+		err = tc.createPrivateKeySecret(false)
+		require.NoError(t, err, "error replacing known private key secret with a random key")
+		// Ensure public key hash and encrypted username annotations are updated correctly
+		err = tc.waitForWindowsNodes(gc.numberOfBYOHNodes, false, false, true)
+		assert.NoError(t, err, "error waiting for Windows nodes updated with newly created private key")
+	})
+
+	// Re-create the known private key so SSH connection can be re-established
+	require.NoError(t, tc.createPrivateKeySecret(true), "error confirming known private key secret exists")
+}
```

Comment:
Shouldn't this be done at the same time the privatekey change is being validated for Machines, using the same new private key?

This will require the reordering of some tests

Reply:
I am not clear on the advantage of doing so (bit quicker/less API call I suppose). The reasoning behind the current separation is that, when priv key changes, BYOH nodes are just patched whereas Machine nodes are deleted and recreated - different test suites in my mind, reconfiguration vs creation.

Reply:
The expected result is different yes, but the advantage is that the setup is the same, so overall testing time is saved, as well as less duplication occuring in the codebase so the tests are easier to understand. You can structure the test like so:
```
func testPrivateKeyChange(t *testing.T) {
  changeKey()
  t.Run("testing expected BYOH results", testFunction)
  t.Run("testing expected Machine results", testFunction)
}
```

Reply:
+1

Reply:
I tried my hand at this, please let me know if my implementation captures the ideas discussed here.